src/Periphery/ServiceFeeCollector.sol

```diff
@@ -0,0 +1,144 @@
+
+    /// External Methods ///
+
+    /// @notice Collects gas fees
+    /// @param tokenAddress The address of the token to collect
+    /// @param feeAmount The amount of fees to collect
+    /// @param receiver The address to send gas to on the destination chain
+    function collectTokenGasFees(
+        address tokenAddress,
+        uint256 feeAmount,
+        address receiver
+    ) external {
+        LibAsset.depositAsset(tokenAddress, feeAmount);
+        emit GasFeesCollected(tokenAddress, receiver, feeAmount);
+    }
+
+    /// @notice Collects gas fees in native token
+    /// @param feeAmount The amount of native token to collect
+    /// @param receiver The address to send gas to on destination chain
+    function collectNativeGasFees(uint256 feeAmount, address receiver)
+        external
+        payable
+    {
+        if (msg.value < feeAmount) revert NotEnoughNativeForFees();
+        uint256 remaining = msg.value - (feeAmount);
+        // Prevent extra native token from being locked in the contract
+        if (remaining > 0) {
+            (bool success, ) = payable(msg.sender).call{ value: remaining }(
+                ""
+            );
+            if (!success) {
+                revert TransferFailure();
+            }
+        }
+        emit GasFeesCollected(LibAsset.NULL_ADDRESS, receiver, feeAmount);
+    }
+
+    /// @notice Collects insurance fees
+    /// @param tokenAddress The address of the token to collect
+    /// @param feeAmount The amount of fees to collect
+    /// @param receiver The address to insure
+    function collectTokenInsuranceFees(
+        address tokenAddress,
+        uint256 feeAmount,
+        address receiver
+    ) external {
+        LibAsset.depositAsset(tokenAddress, feeAmount);
+        emit InsuranceFeesCollected(tokenAddress, receiver, feeAmount);
+    }
+
+    /// @notice Collects insurance fees in native token
+    /// @param feeAmount The amount of native token to collect
+    /// @param receiver The address to insure
+    function collectNativeInsuranceFees(uint256 feeAmount, address receiver)
+        external
+        payable
+    {
+        if (msg.value < feeAmount) revert NotEnoughNativeForFees();
+        uint256 remaining = msg.value - (feeAmount);
+        // Prevent extra native token from being locked in the contract
+        if (remaining > 0) {
+            (bool success, ) = payable(msg.sender).call{ value: remaining }(
+                ""
+            );
+            if (!success) {
+                revert TransferFailure();
+            }
+        }
+        emit InsuranceFeesCollected(
+            LibAsset.NULL_ADDRESS,
+            receiver,
+            feeAmount
+        );
+    }
+
+    /// @notice Withdraws fees
+    /// @param tokenAddress The address of the token to withdraw fees for
+    function withdrawFees(address tokenAddress) external onlyOwner {
+        uint256 balance = LibAsset.getOwnBalance(tokenAddress);
+        if (balance == 0) {
```

Comment:
I dont think such a check is required in an admin-only function. If balance is 0, it will just fail but we can trust (ourselves) to know what we are doing and save deployment gas here